ControlRoomApplication/ControlRoomApplication/Entities/RadioTelescope/RadioTelescopeConfig.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+
+namespace ControlRoomApplication.Entities
+{
+    /// <summary>
+    /// Data class for serializing and deserializing JSON strings into
+    /// RadioTelescopeConfig objects, used to tell the ControlRoom software
+    /// which telescope to run from the database given an ID.
+    /// 
+    /// If no ID is given, the newTelescope flag is set which tells the 
+    /// ControlRoom to create a new one.
+    /// </summary>
+    public class RadioTelescopeConfig
+    {
+        private static string FILE_NAME = "RadioTelescopeConfig.json";
+        public static string DEFAULT_JSON_CONTENTS = "{\"telescopeID\":0,\"newTelescope\":true}";
+        public int telescopeID;
+        public bool newTelescope;
+
+        /// <contains>An integer representing the ID of a RadioTelescope instance to be retrived from the database</contains>
+        /// <contains>A boolean value to represent whether or not a new telescope should be created </contains>
+        public RadioTelescopeConfig(int telescopeID, bool newTelescope)
+        {
+            this.telescopeID = telescopeID;
+            this.newTelescope = newTelescope;
+        }
+
+
+        /// <summary>
+        /// Method to deserialize the output of the JSON file into a RadioTelescopConfig object. 
+        /// File.ReadAllText is guranteed to close the file.
+        /// </summary>
+        /// <returns>null, if any errors occur in parsing or creating a new RT instance. Otherwise, it will return an instance of an RT</returns>
+        public static RadioTelescopeConfig DeserializeRTConfig()
+        {
+            string fileContents;
+            try
+            {
+                fileContents = File.ReadAllText(FILE_NAME);
+            }catch(FileNotFoundException e)
+            {
+                CreateAndWriteToNewJSONFile(DEFAULT_JSON_CONTENTS);
+                fileContents = File.ReadAllText(FILE_NAME);
+            }
+            RadioTelescopeConfig RTConfig;
+
+            // check to make sure JSON file contains valid JSON
+            try
+            {
+                JObject.Parse(fileContents);
+                RTConfig = JsonConvert.DeserializeObject<RadioTelescopeConfig>(fileContents);
+            }
+            // file is either corrupted or contains invalid JSON. Create a new file if this is the case using the default contents
+            catch(JsonReaderException e)
+            {
+                Console.WriteLine(e);
+                RadioTelescopeConfig.CreateAndWriteToNewJSONFile(DEFAULT_JSON_CONTENTS);
```

Comment:
This is still being executed if the user chooses to not recreate the JSON, and it's being executed twice if they _do_. This extra call can likely be safely removed.